// containers/Quadcopter/index.jsx
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import * as R from 'ramda';

import { 
    fetchQuadcopterById,
    addQuadcopterToBasket
} from 'actions';
import { getQuadcopterById } from 'selectors';

import BasketCart from 'components/BasketCart';
import Preloader from 'components/Preloader';

class Quadcopter extends Component {
    componentDidMount() {
        this.props.fetchQuadcopterById(this.props.match.params.id)
    }

    renderContent() {
        const { quadcopter } = this.props;
        return(
            <div className="quad-container img-thumbnail">
                <div className="row">
                    <div className="col-md-6">
                        <img 
                            src={quadcopter.image}
                            alt={quadcopter.name}
                        />
                    </div>
                    <div className="col-md-6">
                        {this.renderFields()}
                    </div>
                </div>
                <div className="figure-caption">
                    <h4 className="float-right">${quadcopter.price}</h4>
                    <h4>{quadcopter.name}</h4>
                    <p>{quadcopter.description}</p>
                </div>
            </div>
        )
    }

    renderFields() {
        const { quadcopter } = this.props;
        const columnFields = R.compose(
            R.toPairs,
            R.pick([
                "flightTime",
                "maxRange",
                "diagonalSize",
                "maxVideo",
                "GPS",
                "GLONASS",
                "flips",
                "intelligentFlight"
            ])
        )(quadcopter)

        return columnFields.map(([key, value]) => (
            <div className="column" key={key}>
                <div className="ab-details-title">
                    <p>{key}</p>
                </div>
                <div className="ab-details-info">
                    {value}
                </div>
            </div>
        ) )
     }

    renderSidebar() {
        const { quadcopter, addQuadcopterToBasket } = this.props;
        return(
            <div className="quadcopter-sidebar">
                <h3 className="lead">Quick shop</h3>
                <BasketCart />
                <div className="form-group">
                    <h1>{quadcopter.name}</h1>
                    <h2>${quadcopter.price}</h2>
                </div>
                <Link 
                    to="/" 
                    className="btn btn-info btn-block"
                >
                    Back to store
                </Link>
                <button 
                    type="button" 
                    className="btn btn-success btn-block"
                    onClick={() => addQuadcopterToBasket(quadcopter.id)}
                >
                    Add to cart
                </button>
            </div>
        )
    }

    render() {
        const { quadcopter, fetching } = this.props;
        return (
            <div className="view-container">
                <div className="container">
                    <div className="row">
                        <div className="col-md-9">
                            {
                                fetching ?
                                <Preloader /> :
                                quadcopter && this.renderContent()
                            }
                        </div>
                        <div className="col-md-3">
                            {quadcopter && this.renderSidebar()}
                        </div>
                    </div> 
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => ({
    quadcopter: getQuadcopterById(state, state.quadcopterPage.id),
    fetching: state.quadcopterPage.fetching
})

const saveActionsToProps = {
    fetchQuadcopterById,
    addQuadcopterToBasket
}

export default connect(mapStateToProps, saveActionsToProps)(Quadcopter);
