// containers/Quadcopters/index.jsx
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import { compose } from 'redux';

import { 
    fetchQuadcopters, 
    loadMoreQuadcopters,
    addQuadcopterToBasket,
    fetchCategories
} from 'actions';
import { getQuadcopters } from 'selectors';
import { shortenString } from 'helpers/helpers';

import Preloader from 'components/Preloader';

class Quadcopters extends Component {

    componentDidMount() {
        this.props.fetchQuadcopters();
        this.props.fetchCategories()
    }

    renderQuadcopter(quadcopter, index) {
        const { addQuadcopterToBasket } = this.props;
        const shortDescription = shortenString(quadcopter.description, 60);
        return (
            <div className="col-12 col-sm-6 col-md-4 col-lg-4 book-list" key={index}>
                <div className="quad-container img-thumbnail">
                    <img 
                        src={quadcopter.image} 
                        alt={quadcopter.name} 
                    />
                    <div className="figure-caption">
                        <h4 className="float-right">${quadcopter.price}</h4>
                        <h4>
                            <Link to={`/quadcopters/${quadcopter.id}`}>
                                {quadcopter.name}
                            </Link>
                        </h4>
                        <p>{shortDescription}</p>
                        <p className="itemButton">
                            <button 
                                className="btn btn-primary"
                                onClick={() => addQuadcopterToBasket(quadcopter.id)}
                            >
                                Buy now!
                            </button>
                            <Link 
                                to={`/quadcopters/${quadcopter.id}`} 
                                className="btn btn-default"
                            >
                                More Info
                            </Link>
                        </p>
                    </div>
                </div>
            </div>
        )
    }

    render() {
        const { quadcopters, fetching, loadMoreQuadcopters } = this.props;
        return  (
            <div>   
                <div className="books row">
                    {
                        fetching ? 
                        <Preloader /> : 
                        quadcopters.map( (quadcopter, index) => this.renderQuadcopter(quadcopter, index) )
                    }
                </div>
                <div className="row">
                    <div className="col-md-12 load-more">
                        <button
                            className="float-right btn btn-outline-primary"
                            onClick={loadMoreQuadcopters}
                        >
                        Load more
                        </button>
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => ({
    quadcopters: getQuadcopters(state, ownProps),
    fetching: state.quadcoptersPage.fetching
})

const saveActionsToProps = {
    fetchQuadcopters,
    loadMoreQuadcopters,
    addQuadcopterToBasket,
    fetchCategories
}

export default compose(
    withRouter,
    connect(mapStateToProps, saveActionsToProps)
)(Quadcopters);
