// api/mockQuadcopters.js
export default [
    {
        id: "1",
        categoryId: "1",
        name: "DJI Spark",
        description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
        price: 500,
        image: "/uploads/DJI_Spark.jpg",
        flightTime: "16 min",
        maxRange: "100 m",
        maxSpeed: "35 kmh",
        weight: "300 g",
        diagonalSize: "170 mm",
        maxPhoto: "3968x2976 px",
        maxVideo: "1920x1080 @30fps",
        GPS: "yes",
        GLONASS: "yes",
        flips: "no",
        intelligentFlight: "yes"
    },
    {
        id: "2",
        categoryId: "2",
        name: "Hubsan X4 H502S",
        description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
        price: 180,
        image: "/uploads/Hubsan_X4_H502S.png",
        flightTime: "13 min",
        maxRange: "300 m",
        maxSpeed: "",
        weight: "147 g",
        diagonalSize: "230 mm",
        maxPhoto: "",
        maxVideo: "1280x720 @60fps",
        GPS: "yes",
        GLONASS: "no",
        flips: "",
        intelligentFlight: "no"
    },
    {
        id: "3",
        categoryId: "3",
        name: "Syma X8SW",
        description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
        price: 55,
        image: "/uploads/Syma_X8SW.jpg",
        flightTime: "9 min",
        maxRange: "70 m",
        maxSpeed: "",
        weight: "",
        diagonalSize: "707 mm",
        maxPhoto: "1280x720 px",
        maxVideo: "1280x720",
        GPS: "no",
        GLONASS: "no",
        flips: "no",
        intelligentFlight: "no"
    },
    {
        id: "4",
        categoryId: "1",
        name: "DJI Phantom 4 Pro",
        description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
        price: 1500,
        image: "/uploads/DJI_Phantom4_pro.jpg",
        flightTime: "30 min",
        maxRange: "7000 m",
        maxSpeed: "70 kmh",
        weight: "1380 g",
        diagonalSize: "350 mm",
        maxPhoto: "5472x3648 px",
        maxVideo: "4096x2160 @50fps",
        GPS: "yes",
        GLONASS: "yes",
        flips: "no",
        intelligentFlight: "yes"
    },
    {
        id: "5",
        categoryId: "2",
        name: "Hubsan X4 H501S",
        description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
        price: 230,
        image: "/uploads/Hubsan_H501S_X4.jpg",
        flightTime: "20 min",
        maxRange: "300 m",
        maxSpeed: "",
        weight: "410 g",
        diagonalSize: "311 mm",
        maxPhoto: "",
        maxVideo: "1920x1080 @30fps",
        GPS: "yes",
        GLONASS: "no",
        flips: "no",
        intelligentFlight: "yes"
    },
    {
        id: "6",
        categoryId: "3",
        name: "Syma X8HG",
        description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
        price: 130,
        image: "/uploads/Syma_X8HG.jpg",
        flightTime: "6 min",
        maxRange: "70 m",
        maxSpeed: "",
        weight: "",
        diagonalSize: "700 mm",
        maxPhoto: "3264x2448 px",
        maxVideo: "1280x720",
        GPS: "no",
        GLONASS: "no",
        flips: "yes",
        intelligentFlight: "no"
    },
    {
        id: "7",
        categoryId: "3",
        name: "Syma X5UW",
        description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
        price: 85,
        image: "/uploads/Syma_X5UW.jpg",
        flightTime: "7 min",
        maxRange: "70 m",
        maxSpeed: "",
        weight: "",
        diagonalSize: "450 mm",
        maxPhoto: "",
        maxVideo: "1280x720",
        GPS: "no",
        GLONASS: "no",
        flips: "yes",
        intelligentFlight: "yes"
    },
    {
        id: "8",
        categoryId: "4",
        name: "Parrot Bebop 2",
        description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
        price: 380,
        image: "/uploads/Parrot_Bebop2.jpg",
        flightTime: "25 min",
        maxRange: "2000 m",
        maxSpeed: "60 kmh",
        weight: "500 g",
        diagonalSize: "",
        maxPhoto: "",
        maxVideo: "1920x1080 @30fps",
        GPS: "yes",
        GLONASS: "yes",
        flips: "no",
        intelligentFlight: "yes"
    }
]
